### DISCUSSION TITLE
Help with modding paintings 1.18.1

### DISCUSSION THREAD

**Discussion by Dog517** [FabricMC/fabric-api#2135] [Mod Dev Support]
Hello, so I want to make some paintings for fabric 1.18.1, so I followed a tutorial and set everything up, but in the game when i put down my painting it shows the missing texture. here is my ModPaintings class :

`package net.dogdev.stuffmod.painting;

import net.dogdev.stuffmod.StuffMod;
import net.fabricmc.fabric.api.networking.v1.S2CPlayChannelEvents;
import net.minecraft.entity.decoration.painting.PaintingMotive;
import net.minecraft.util.Identifier;
import net.minecraft.util.registry.Registry;

public class ModPaintings {

    public static final PaintingMotive RICK = RegisterPaintings("rick", new PaintingMotive(16,16));

        private static PaintingMotive RegisterPaintings(String name, PaintingMotive paintingMotive){
            return Registry.register(Registry.PAINTING_MOTIVE, new Identifier(StuffMod.MOD_ID, name), paintingMotive);
        }

    public static void registerPainting(){

        StuffMod.LOGGER.info("Registering Paintings For " + StuffMod.MOD_ID);

    }
}
`

and the structure for the textures is this:  resources.assets.stuffmod.textures.painting.rick.png

Thanks!

**Comment by warjort** [comment]
> and the structure for the textures is this: resources.assets.stuffmod.textures.painting.rick.png

If you are following (maven) conventions the png would be in your source tree as
````src/main/resources/assets/stuffmod/textures/painting/rick.png````
assuming the id is ````stuffmod:rick````
this should be in your mod jar as
````assets/stuffmod/textures/painting/rick.png````

You can use ````jar -tf yourmod.jar```` to look in the jar (in your build/libs) or anything else that lets you look in zip files.

Also try running
````
                PaintingMotive motive = // the thing you register
		Sprite sprite = MinecraftClient.getInstance().getPaintingManager().getPaintingSprite(motive);
````
at runtime (after it loads resource packs not at registration time) to see if it actually loaded your image.

**Reply by Dog517** [reply]
I kinda dont understand you completely. my png file is already at assets/stuffmod/textures/painting/rick.png, I am also kinda new at modding so a dumbed-down reply would be best for me.

**Reply by warjort** [reply]
You need to "show not tell". Otherwise we can't see where you have gone wrong.
Just asserting you did it correctly leaves us with nothing to work with.

e.g. I just tried this with 1.18.1 by modifying the fabric example mod:

Show the image in the correct location
````
$ jar -tf build/libs/fabric-example-mod-1.0.0.jar | grep test.png
assets/modid/textures/painting/test.png
````

This is my initializer
````
public class ExampleMod implements ModInitializer {
        // This logger is used to write text to the console and the log file.
        // It is considered best practice to use your mod id as the logger's name.
        // That way, it's clear which mod wrote info, warnings, and errors.
        public static final Logger LOGGER = LogManager.getLogger("modid");

        public static final PaintingMotive PAINTING = new PaintingMotive(16, 16);

        @Override
        public void onInitialize() {
                // This code runs as soon as Minecraft is in a mod-load-ready state.
                // However, some things (like resources) may still be uninitialized.
                // Proceed with mild caution.

                LOGGER.info("Hello Fabric world!");
                Registry.register(Registry.PAINTING_MOTIVE, new Identifier("modid", "test"), PAINTING);
        }
}
````
This worked for me, my test.png is one of the 1x1 block paintings.

**Reply by Dog517** [reply]
Here, I made a Github repository so you can see if im doing anything wrong.

https://github.com/Dog517/fabric-StuffMod-mod-1.18

**Reply by warjort** [reply]
This should be "stuffmod"
https://github.com/Dog517/fabric-StuffMod-mod-1.18/blob/26a5cc1e8978ddae1f79934860defd3e237680c5/src/main/java/net/dogdev/stuffmod/StuffMod.java#L20

**Reply by Dog517** [reply]
> This should be "stuffmod" https://github.com/Dog517/fabric-StuffMod-mod-1.18/blob/26a5cc1e8978ddae1f79934860defd3e237680c5/src/main/java/net/dogdev/stuffmod/StuffMod.java#L20

Thanks! for whatever reason this fixed it!

**Reply by warjort** [reply]
You are using this mod id value as your "namespace" in the identifier when registering the painting.
It must match the "namespace" in your mod's jar otherwise it won't find the file.
Assets have the generic form:
````assets/namespace/thing/name.ext````

Before it was looking for ````assets/tutorialmod/textures/painting/rick.png````

**Reply by Dog517** [reply]
Ah.. That makes sense. Thank you for helping me figure it out!
